### PR TITLE
test-network: add udev related tests for networkctl

### PR DESCRIPTION
**Pull request by yuwata**
WIP.

### CODE REVIEW

**Comment by yuwata** [comment]
Hmm, CentOS CI (Arch in KVM) shows
```
test_udev (__main__.NetworkctlTests) ... 
P: /devices/virtual/net/dummy98
L: 0
E: DEVPATH=/devices/virtual/net/dummy98
E: INTERFACE=dummy98
E: IFINDEX=31
E: SUBSYSTEM=net
E: USEC_INITIALIZED=2364363138
E: ID_NET_NAMING_SCHEME=v243
E: ID_NET_DRIVER=dummy
E: SYSTEMD_ALIAS=/sys/subsystem/net/devices/dummy98
E: TAGS=:systemd:
â— 31: dummy98                                                                   
             Link File: n/a                                                     
          Network File: /run/systemd/network/netdev-link-local-addressing-yes.nâ€¦
                  Type: ether                                                   
                 State: degraded (configured)                                   
                Driver: dummy                                                   
            HW Address: c2:2f:e3:09:3d:de(null)                                 
                   MTU: 1500                                                    
  Queue Length (Tx/Rx): 1/1                                                     
               Address: 169.254.148.221                                         
                        fe80::c02f:e3ff:fe09:3dde
```
Why no `ID_NET_LINK_FILE=` is set???

**Comment by yuwata** [comment]
Still I am not sure why udev does not find .link file. But let's ignore that, as the test is just for `networkctl`.

File: test/test-network/systemd-networkd-tests.py [hunk 1]
```diff
@@ -91,6 +91,23 @@ def f(func):
 
     return f
 
+def expectedFailureIfLinkFileFieldIsNotSet():
+    def f(func):
+        support = False
+        rc = call('ip link add name dummy99 type dummy')
+        if rc == 0:
+            ret = run('udevadm info -w10s /sys/class/net/dummy99', stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```
Comment (poettering) [inline test/test-network/systemd-networkd-tests.py:99]: this is racy, the event from the kernel might not have reached udev yet by the time you wait on it here... 

i wonder if we maybe should introduce "udevadm wait" or so, where you can specify a device and it waits (with a timeout) until the device shows up in the udev db if its's not there. This is a lot more correct then the "settle" stuff people usually try...

Comment (yuwata) [reply test/test-network/systemd-networkd-tests.py:99]: I've already add `-w` option for that purpose.

**Comment by keszybz** [comment]
This seems to have been forgotten. Let's merge it.

**Pull request merged**